lib/modules/datasource/npm/get.ts

```diff
@@ -171,26 +171,17 @@ export async function getDependency(
       return release;
     });
     logger.trace({ dep }, 'dep');
-    // serialize first before saving
-    // TODO: use dynamic detection of public repos instead of a static list (#9587)
-    const whitelistedPublicScopes = [
-      '@graphql-codegen',
-      '@storybook',
-      '@types',
-      '@typescript-eslint',
-    ];
-    if (
-      !raw.authorization &&
-      (whitelistedPublicScopes.includes(packageName.split('/')[0]) ||
-        !packageName.startsWith('@'))
-    ) {
+    if (/(^|,)\s*public\s*(,|$)/.test(raw.headers?.['cache-control'] ?? '')) {
```

Comment:
```suggestion
    const cacheControl = raw.headers?.['cache-control']
    if (is.nonEmptyString(cacheControl) && regEx(/(^|,)\s*public\s*(,|$)/).test(cacheControl)) {
```
We should use our regex utilities and this is IMO more readable.